Library/Homebrew/formula_installer.rb

```diff
@@ -488,9 +488,9 @@ def expand_dependencies(deps)
 
       if dep.prune_from_option?(build)
         Dependency.prune
-      elsif include_test? && dep.test? && !dep.installed?
-        Dependency.keep_but_prune_recursive_deps
-      elsif dep.build? && install_bottle_for?(dependent, build)
+      elsif dep.test? && !dep.build? && !include_test?
```

Comment:
Maybe we want a `(!include_test? || dependent != formula)`? I think we only want to bring in test dependencies if they are immediate dependencies.